TransactionService/docker_context/handlers/TransactionHandler.py

```diff
@@ -0,0 +1,126 @@
+from datetime import datetime
+from math import ceil, floor
+from sanic import Sanic, response
+
+
+def errorResult(err, data):
+    return {
+        'errorMessage': err,
+        'content': data
+    }
+
+
+def goodResult(msg, data):
+    return {
+        'message': msg,
+        'content': data
+    }
+
+
+class TransactionHandler:
+
+    # TO DO:
+    # VERIFICATION OF DATA RECEIVED FROM CACHE?
+    # AUDIT
+    # RECHECK PRICE ON COMMIT
+
+    def __init__(self, audit, client, ip, port):
+        self.audit = audit
+        self.client = client
+        self.cacheURL = f'http://{ip}:{port}'
+
+    async def addFunds(self, trans_num, user_id, amount):
+        data = {'user_id': user_id, 'funds': float(amount)}
+        result = await self.postRequest(self.cacheURL+"/funds/add_funds", data)
+        return result
+
+    async def getQuote(self, trans_num, user_id, stock_id):
+        result = await self.getRequest(self.cacheURL + f"/quote/get/{user_id}/{stock_id}/{trans_num}")
+        # DO AUDIT IN CACHE SERVICE
+        return result
+
+    async def buyStock(self, trans_num, user_id, stock_symbol, amount):
+        quotePrice = await self.getQuote(trans_num, user_id, stock_symbol)
+        totalStock = floor(float(amount)/float(quotePrice))
+        totalValue = quotePrice*totalStock
+        userBal = await self.getRequest(self.cacheURL + f"/funds/get/user/{user_id}")
+        userBal = userBal.data
+        if totalValue > float(userBal):
+            return response.json(errorResult(err="User doesn't have required funds", data=''), status=400)
```

Comment:
think we just want to return errorResult here? since the api is what handles creating the response.json object itself